src/xrpld/rpc/handlers/BookOffers.cpp

```diff
@@ -204,13 +204,13 @@ doBookOffers(RPC::JsonContext& context)
 Json::Value
 doBookChanges(RPC::JsonContext& context)
 {
-    auto res = RPC::getLedgerByContext(context);
+    std::shared_ptr<ReadView const> ledger;
 
-    if (std::holds_alternative<Json::Value>(res))
-        return std::get<Json::Value>(res);
+    Json::Value result = RPC::lookupLedger(ledger, context);
```

Comment:
1. Does this change still yield the same error from checking ledger_index etc.?
2. Can we do the same in `LedgerRequest.cpp` and eliminate getLedgerByContext() entirely?

Reply:
1. @godexsoft the `ledger_index` handling (and errors) will be the same as in almost every other command, since the vast majority of commands use `lookupLedger`
2. No; the ADMIN method `ledger_request` which, with this PR, will be the only method using `getLedgerByContext()`, and it  needs to rely on this function in order to download ledgers from the other nodes.